src/app/dataset/datasetform/datasetform.component.spec.ts

```diff
@@ -71,4 +70,50 @@ describe('DatasetformComponent', () => {
     expect(localStorage.getItem('tempDatasetData')).not.toBe('');
     localStorage.removeItem('tempDatasetData');
   });
+
+  it('should clear the fields', () => {
+    fixture.detectChanges();
+    const fName = 'datasetName';
+    const field = component.datasetForm.controls[fName];
+
+    expect(component.datasetForm.dirty).toBeFalsy();
+    expect(field.value).toBeTruthy();
+
+    component.clearField(fName);
+
+    expect(component.datasetForm.dirty).toBeTruthy();
+    expect(field.value).toBeFalsy();
+  });
+
+  it('should reset', () => {
+    fixture.detectChanges();
+    const fName = 'datasetName';
+    expect(component.datasetForm.pristine).toBeTruthy();
+    component.clearField(fName);
+    expect(component.datasetForm.pristine).toBeFalsy();
+    component.reset();
+    expect(component.datasetForm.pristine).toBeTruthy();
+  });
+
+  it('should add redirection ids', () => {
+    fixture.detectChanges();
+    const existingId = '123';
+    const newId = 'some_id';
+    expect(component.datasetForm.dirty).toBeFalsy();
+    component.addRedirectionId(existingId);
+    expect(component.datasetForm.dirty).toBeFalsy();
+    component.addRedirectionId(newId);
+    expect(component.datasetForm.dirty).toBeTruthy();
+  });
+
+  it('should remove redirection ids', () => {
+    fixture.detectChanges();
+    const existingId = '123';
+    const newId = 'some_id';
```

Comment:
As these are repeated (here and in the previous scenario), could you move them out of the "it" and define them just once?

Reply:
Done